src/convertToObject.js

```diff
@@ -6,7 +6,21 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const obj = {};
+
+  sourceString
+    .trim()
+    .split(';')
+    .map((item) => item.trim().split(':'))
+    .map((item) => item.map((mapitem) => mapitem.trim()))
+    .filter((item) => item && item.length > 1)
```

Comment:
I would combine these checks under the `foreach` call

The reason is, that each `map` and `filter` call is iterating the whole collection

Looks like you are iterating more times than needed